src/rt/memset.d

```diff
@@ -100,10 +100,21 @@ real *_memset80(real *p, real value, size_t count)
     return pstart;
 }
 
-c_complex_real *_memset160(c_complex_real *p, c_complex_real value, size_t count)
+static if (is(c_long_double == real))
+    private alias d_complex_real = c_complex_real;
+else
+{
+    private struct d_complex_real
+    {
+        real re;
+        real im;
+    }
+}
+
+d_complex_real *_memset160(d_complex_real *p, d_complex_real value, size_t count)
```

Comment:
Why add all the boilerplate for what is essentially just:
```suggestion
_Complex!real *_memset160(_Complex!real *p, _Complex!real value, size_t count)
```

Reply:
`_Complex(T)` is private.

Reply:
A `package(rt)` declaration of it could be copied to `rt.util.utility` with aliases for d_cfloat, d_cdouble, and d_creal.

Reply:
A one-liner `private struct d_complex_real { real re, im; }` would be another alternative. I really don't want to extract this struct or template anywhere just to reference it in some DMD-specific file and save some lines.

Reply:
It'd be used by rt.typeinfo, so no, it won't be restricted to a dmd-specific file.

Reply:
You'll need to go through the enum interface anyway, otherwise it may not have the correct ABI.

Reply:
> It'd be used by rt.typeinfo, so no, it won't be restricted to a dmd-specific file.

For DMD only, and only for about ~20 months until we can remove this crap after the deprecation is over.

> You'll need to go through the enum interface anyway, otherwise it may not have the correct ABI.

[DMD only again.] Ouch, I assumed it's all good because all tests still pass. - My motivation just sank to 0.

Reply:
>  My motivation just sank to 0.

I'll own this then, since I introduced this "regression".  Will leave you just to deal with the core/stdc side.

Reply:
Done, please rebase @kinke.